pkgs/test/nixpkgs-check-by-name/README.md

```diff
@@ -42,9 +29,20 @@ These checks are performed by this tool:
 
 ### Nix evaluation checks
 - `pkgs.${name}` is defined as `callPackage pkgs/by-name/${shard}/${name}/package.nix args` for some `args`.
-  - **Only after --version v1**: If `pkgs.${name}` is not auto-called from `pkgs/by-name`, `args` must not be empty
 - `pkgs.lib.isDerivation pkgs.${name}` is `true`.
 
+### Ratchet checks
+
+Furthermore, this tool implements certain [ratchet](https://qntm.org/ratchet) checks.
+This allows gradually phasing out deprecated patterns without breaking the base branch or having to migrate it all at once.
+It works by not allowing new instances of the pattern to be introduced, but allowing already existing instances.
+The existing instances are coming from `<BASE_NIXPKGS>`, which is then checked against `<NIXPKGS>` for new instances.
+Ratchets should be removed eventually once the pattern is not used anymore.
+
+The current ratchets are:
+
+- If `pkgs.${name}` is not auto-called from `pkgs/by-name`, the `args` in its `callPackage` must not be empty,
```

Comment:
That's hardly intelligible.

In part, due to that it's hard to observe an auto-called package, it's automatic after all. So how does a non-auto called look like?

I don't know what exactly is meant, but maybe could be:

> It is not allowed to introduce new `callPackage` calls in `top-level.nix`, if those would have empty `args`.

Reply:
While it's not strictly necessary for users to understand these checks (that's what CI is for!), it's still good for future developers. Pushed https://github.com/NixOS/nixpkgs/pull/272395/commits/fc2d26939d2839d65553e8ca605f2440a45fc387 to improve this!